File: src/main/java/com/coreyd97/BurpExtenderUtilities/VariableViewPanel.java
package com.coreyd97.BurpExtenderUtilities;

import javax.swing.*;
import java.awt.*;

/**
 * Created by corey on 24/08/17.
 */
public class VariableViewPanel extends JPanel {
    public enum View {HORIZONTAL, VERTICAL, TABS}
    private final Component a;
    private final String aTitle;
    private final Component b;
    private final String bTitle;
    private Preferences preferences;
    private String preferenceKey;
    private Component wrapper;
    private View view;

    public VariableViewPanel(Preferences prefs, String preferenceKey,
                             Component a, String aTitle, Component b, String bTitle,
                             View defaultView){
        this.preferences = prefs;
        this.preferenceKey = preferenceKey;

        if(this.preferences != null && this.preferenceKey != null){
            this.preferences.registerSetting(this.preferenceKey, View.class, defaultView, Preferences.Visibility.GLOBAL);
            view = (View) this.preferences.getSetting(this.preferenceKey);
        }else{
            view = defaultView;
        }

        this.a = a;
        this.aTitle = aTitle;
        this.b = b;
        this.bTitle = bTitle;
        this.setLayout(new BorderLayout());
        this.setView(view);
    }

    public VariableViewPanel(Preferences preferences, String preferenceKey,
                             Component a, String aTitle, Component b, String bTitle) {
        this(preferences, preferenceKey, a, aTitle, b, bTitle, null);
    }

    public VariableViewPanel(Component a, String aTitle, Component b, String bTitle, View defaultView){
        this(null, null, a, aTitle, b, bTitle, defaultView);
    }

    public View getView(){
        return this.view;
    }

    public void setView(View view){
        if(view == null) view = View.VERTICAL;
        switch (view){
            case HORIZONTAL:
            case VERTICAL: {
                this.wrapper = new JSplitPane();
                ((JSplitPane) wrapper).setLeftComponent(a);
                ((JSplitPane) wrapper).setRightComponent(b);
                if(view == View.HORIZONTAL){
                    ((JSplitPane) wrapper).setOrientation(JSplitPane.HORIZONTAL_SPLIT);
                }else{
                    ((JSplitPane) wrapper).setOrientation(JSplitPane.VERTICAL_SPLIT);
                }
                ((JSplitPane) wrapper).setResizeWeight(0.5);
                break;
            }
            case TABS: {
                this.wrapper = new JTabbedPane();
                ((JTabbedPane) wrapper).addTab(aTitle, a);
                ((JTabbedPane) wrapper).addTab(bTitle, b);
                break;
            }
        }
        this.removeAll();
        this.add(wrapper, BorderLayout.CENTER);
        this.revalidate();
        this.repaint();
        this.view = view;

        if(this.preferences != null && this.preferenceKey != null) {
            this.preferences.setSetting(this.preferenceKey, view);
        }
    }



}

File: src/main/java/com/coreyd97/montoyautilities/VariableViewPanel.kt
package com.coreyd97.montoyautilities

import java.awt.BorderLayout
import java.awt.Component
import javax.swing.JPanel
import javax.swing.JSplitPane
import javax.swing.JTabbedPane

/**
 * Created by corey on 24/08/17.
 */
class VariableViewPanel(
    preferenceKey: String,
    private val a: Component, private val aTitle: String, private val b: Component, private val bTitle: String,
    defaultView: View = View.VERTICAL
) : JPanel() {
    enum class View {
        HORIZONTAL, VERTICAL, TABS
    }

    private var wrapper: Component? = null
    private var _view: View by Preference(preferenceKey, defaultView)
    public var view: View = _view
        get() = _view
        set(value) {
            field = value
            updateView(value)
        }

    init {
        this.layout = BorderLayout()
        this.updateView(_view)
    }

    private fun updateView(view: View) {
        when (view) {
            View.HORIZONTAL, View.VERTICAL -> {
                this.wrapper = JSplitPane()
                (wrapper as JSplitPane).leftComponent = a
                (wrapper as JSplitPane).rightComponent = b
                if (view == View.HORIZONTAL) {
                    (wrapper as JSplitPane).orientation = JSplitPane.HORIZONTAL_SPLIT
                } else {
                    (wrapper as JSplitPane).orientation = JSplitPane.VERTICAL_SPLIT
                }
                (wrapper as JSplitPane).resizeWeight = 0.5
            }

            View.TABS -> {
                this.wrapper = JTabbedPane()
                (wrapper as JTabbedPane).addTab(aTitle, a)
                (wrapper as JTabbedPane).addTab(bTitle, b)
            }
        }
        this.removeAll()
        this.add(wrapper, BorderLayout.CENTER)
        this.revalidate()
        this.repaint()
        this._view = view
    }
}
